feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/session_proxy_relay.go

```diff
@@ -44,60 +45,85 @@ func (s *RelayRouter) UpdateSession(
 	// Group requests by PLMNID
 	// Use the longest possible PLMN ID (6), in the worst case we'll fragment requests more then needed, but the routing
 	// will still be logically correct
-	reqMap := map[string]*protos.UpdateSessionRequest{}
+	reqMap := map[string]map[gateway_registry.GwServiceType]*protos.UpdateSessionRequest{}
 	for _, u := range r.GetUpdates() {
 		if u == nil {
 			continue
 		}
 		plmnid := getPlmnId6(u.GetCommonContext().GetSid().GetId())
-		req, ok := reqMap[plmnid]
+		pMap, ok := reqMap[plmnid]
+		if !ok || pMap == nil {
+			pMap = map[gateway_registry.GwServiceType]*protos.UpdateSessionRequest{}
+			reqMap[plmnid] = pMap
+		}
```

Comment:
you need to encapsulate that. You are using the same code in two places

Check the code i added previously on my comments

Reply:
@uri200 , Got it. I thought to initialize the inner map dynamically than statically, if the request contains specific RATType. So it was not encapsulated in the common function. Now I placed the common function to initialize the inner map for GwServiceType to avoid code redundancy. Please check